test/features/signup/signup_screen_test.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:interactive_diary/bloc/authentication/signup/google_signup_bloc.dart';
+import 'package:interactive_diary/features/signup/signup_screen.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../mock_firebase.dart';
+import '../../widget_tester_extension.dart';
+import 'signup_screen_test.mocks.dart';
+
+
+@GenerateMocks(<Type>[GoogleSignupBloc])
+void main() {
+  setupFirebaseAuthMocks();
+  final GoogleSignupBloc mockSignUpBloc = MockGoogleSignupBloc();
+  late IDSignUp screen;
+
+  setUpAll(() async {
+    await Firebase.initializeApp();
+    screen = const IDSignUp();
+  });
+
+  group('Signup with google', () {
+    testWidgets('When State is GoogleSigninInitialState, '
+      'then the Google button is showed with idle status',
+      (WidgetTester widgetTester) async {
+
+        when(mockSignUpBloc.stream)
+            .thenAnswer((_) => Stream<GoogleSignupState>.value(
+            GoogleSignupInitialState()));
+        when(mockSignUpBloc.state)
+          .thenAnswer((_) => GoogleSignupInitialState());
+
+        final Finder googleButton = find.widgetWithText(
+          IDGoogleSignInButton,
+          'Continue with Google'
+        );
+
+        await widgetTester.blocWrapAndPump<GoogleSignupBloc>(mockSignUpBloc, screen, infiniteAnimationWidget: true);
```

Comment:
`IDSignUp` screen has its own `GoogleSignupBloc`, so `blocWrapAndPump` does not provide mocked bloc to it.

Mocking `GoogleSignupBloc` is not needed here if you're only checking existence of `googleButton`